src/core/utils/urlutils.js

```diff
@@ -82,3 +84,66 @@ export function equivalentParams (params1, params2) {
   }
   return true;
 }
+
+/**
+ * Creates a new params object without params that begin with the given prefixes
+ * @param {SearchParams} params The parameters to remove from
+ * @param {string[]} prefixes The prefixes of parameters to remove
+ * @return {SearchParams} A new instance of SearchParams without removed params
+ *   from the params parameter
+ */
+export function removeParamsWithPrefixes (params, prefixes) {
+  const newParams = new SearchParams();
+  for (const [key, val] of params.entries()) {
+    let includeEntry = true;
+    for (const prefix of prefixes) {
+      if (key.startsWith(prefix)) {
+        includeEntry = false;
+      }
+    }
+    if (includeEntry) {
+      newParams.set(key, val);
+    }
+  }
+  return newParams;
+}
+
+/**
+ * Removes parameters for filters, facets, sort options, and pagination
+ * from the params provided. This is useful for constructing
+ * inter-experience answers links.
+ * @param {SearchParams} params The parameters to remove from
+ * @param {function} getComponentNamesForComponentTypes Given string[]
+ *   component types, returns string[] component names for those types
+ * @return {SearchParams} Parameters that have filtered out params that
+ *   should not persist across the answers experience
+ */
+export function filterParamsForExperienceLink (
+  params,
+  getComponentNamesForComponentTypes
```

Comment:
I think this method can probably go into core.js. I think this is one case where it makes less sense to have this as a static method, because it depends on this componentManager, which lives inside of core. Passing in the method as a param is definitely cool and flexible but maybe not necessary here

Reply:
This is an interesting thought. Are you proposing that we move the entire function into componentManager or `getFilterPrefixes` maybe? It _feels_ like it is not the job of the component manager to generate these prefixes but I'm also not opposed to moving it.

Thoughts? @tmeyer2115 @alexisgrow

Reply:
Adding a `getFilterPrefixes` method to the ComponentManager seems outside its responsibility. Its role is to essentially be a registry of the components on the page. It adds them, deletes them, etc. The `getFilterPrefixes` method feels related to a different concern (the construction of cross-page links) and on a different level of abstraction.